internal/librarian/generate.go

```diff
@@ -204,9 +208,18 @@ func (r *generateRunner) runGenerateCommand(ctx context.Context, outputDir strin
 			RepoDir:   r.repo.Dir,
 		}
 		slog.Info("Performing refined generation for library", "id", libraryID)
-		return libraryID, r.containerClient.Generate(ctx, generateRequest)
+		if err := r.containerClient.Generate(ctx, generateRequest); err != nil {
+			return "", err
+		}
+
+		// Push the config changes to GitHub.
+		if err := commitAndPush(ctx, r.repo, r.ghClient, r.cfg.PushConfig); err != nil {
+			return "", err
+		}
```

Comment:
iiuc, we need to check on condition that PushConfig is provided to commit and push.